flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -163,16 +163,21 @@ public static String getTaskManagerStartCommand(
 			String mainClass,
 			@Nullable String mainArgs) {
 		final TaskExecutorResourceSpec taskExecutorResourceSpec = tmParams.getTaskExecutorResourceSpec();
+		final String jvmMemOpts = TaskExecutorResourceUtils.generateJvmParametersStr(taskExecutorResourceSpec);
+		String args = TaskExecutorResourceUtils.generateDynamicConfigsStr(taskExecutorResourceSpec);
+		if (mainArgs != null) {
+			args += " " + mainArgs;
+		}
 		return getCommonStartCommand(
 			flinkConfig,
 			ClusterComponent.TASK_MANAGER,
-			TaskExecutorResourceUtils.generateJvmParametersStr(taskExecutorResourceSpec),
+			jvmMemOpts,
 			configDirectory,
 			logDirectory,
 			hasLogback,
 			hasLog4j,
 			mainClass,
-			mainArgs
+			args
```

Comment:
This is unrelated to this PR but maybe one could think about consolidating the methods `KubernetesUtils#getCommonStartCommand` and `BootstrapTool#getTaskManagerShellCommand`. They look very similar.